File: problems/3100_water_bottles_ii/3100_water_bottles_ii.java
class Solution {
    public int maxBottlesDrunk(int numBottles, int numExchange) {
        int drank = numBottles;
        int empties = numBottles; // 初始空瓶數（喝掉所有瓶子後變為空瓶）
        while (empties >= numExchange) {
            // 用空瓶兌換 + 喝掉新瓶後的效果
            empties = empties - numExchange + 1;
            numExchange++;
            drank++;
        }
        return drank;
    }
}

File: problems/3100_water_bottles_ii/3100_water_bottles_ii.py
class Solution:
    def maxBottlesDrunk(self, numBottles: int, numExchange: int) -> int:
        drank = numBottles
        empties = numBottles
        while empties >= numExchange:
            empties = empties - numExchange + 1
            numExchange += 1
            drank += 1
        return drank
